### ISSUE TITLE
False positive when using local keyword

### ISSUE THREAD

**Issue by baggepinnen** [julia-vscode/StaticLint.jl#349]
The following is a valid piece of julia code that should not be warned against
```julia
function fun()
    local inner_fun
    let
        inner_fun(x) = x
    end
end
```
![image](https://user-images.githubusercontent.com/3797491/193410115-4116b69d-5ecc-44a0-9b69-129cb0b4662e.png)
